frame/util/bli_util_unb_var1.c

```diff
@@ -1085,10 +1085,39 @@ void PASTEMAC(ch,varname) \
 		PASTEMAC2(ch,chr,gets)( *chi1, chi1_r, chi1_i ); \
 \
 		abs_chi1_r = bli_fabs( chi1_r ); \
+		abs_chi1_r = bli_fabs( chi1_i ); \
+\
+		if ( bli_isnan( abs_chi1_r ) ) \
+		{ \
+			sumsq_r = abs_chi1_r; \
+			scale_r = one_r; \
+			break; \
+		} \
+\
+		if ( bli_isnan( abs_chi1_i ) ) \
+		{ \
+			sumsq_r = abs_chi1_i; \
+			scale_r = one_r; \
+			break; \
+		} \
+\
+		if ( bli_isinf( abs_chi1_r ) ) \
+		{ \
+			sumsq_r = abs_chi1_r; \
+			scale_r = one_r; \
+			break; \
```

Comment:
Assuming that in a vector x we have x[0] = inf and x[1] = NaN, then we break out of the for loop and return inf in sumsq_r. From the description in the commit message, I expected to return NaN. Could you please clarify which one is the correct return value?

Reply:
This was fixed in 92b9af48871ee85e831939243c467b9ba5250fdb.

Reply:
In that case, since we do not break in case of NaN, then for x[0] = inf , x[1] = NaN, x[2]= inf, won't we overwrite sumsq_r with inf again?

Reply:
There is a `continue` in the most recent version ([here](https://github.com/flame/blis/blob/92b9af48871ee85e831939243c467b9ba5250fdb/frame/util/bli_util_unb_var1.c#L1106)). Once a NaN is encountered this will trigger every iteration. That particular line could be a `break` but it doesn't make a difference to the result.